### PR TITLE
Flake8 update fix

### PR DESCRIPTION
**Pull request by williamjameshandley**
# Description

The CI is failing due to a recent upgrade in flake8. This replaces the appropriate `==` comparison with `is` for testing types

# Checklist:

- [X] I have performed a self-review of my own code
- [X] My code is PEP8 compliant (`flake8 anesthetic tests`)
- [X] My code contains compliant docstrings (`pydocstyle --convention=numpy anesthetic`)
- [X] New and existing unit tests pass locally with my changes (`python -m pytest`)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have appropriately incremented the [semantic version number](https://semver.org/) in both README.rst and anesthetic/_version.py

### CODE REVIEW

File: anesthetic/plot.py [hunk 1]
```diff
@@ -969,7 +969,7 @@ def hist_plot_1d(ax, data, *args, **kwargs):
     xmax = quantile(data, q[-1], weights)
     range = kwargs.pop('range', (xmin, xmax))
 
-    if type(bins) == str and bins in ['knuth', 'freedman', 'blocks']:
+    if type(bins) is str and bins in ['knuth', 'freedman', 'blocks']:
```
Comment (lukashergt) [inline anesthetic/plot.py:972]: I believe the more canonical way is actually to use `isintance(bins, str)`.

File: tests/test_reader.py [hunk 2]
```diff
@@ -281,14 +281,14 @@ def test_hdf5(tmp_path, root):
 
     with HDFStore(filename) as store:
         assert_frame_equal(samples, store[key])
-        assert type(store[key]) == type(samples)
+        assert type(store[key]) is type(samples)
```
Comment (lukashergt) [inline tests/test_reader.py:284]: Possibly `isinstance(store[key], type(samples))` or `isinstance(store[key], Samples)` would be more appropriate. `type(.) is type(.)` is stricter. Not sure which would be better here and in the rest of this test...

**Review by lukashergt — COMMENTED** [review]
See comments inline.

File: anesthetic/plot.py (hunk 1 above)
Comment (williamjameshandley) [reply anesthetic/plot.py]: agreed. Updated in c6d173e

File: tests/test_reader.py (hunk 2 above)
Comment (williamjameshandley) [reply tests/test_reader.py:284]: stricter is better here -- we definitely want the type to not change in between writing and reading.

**Review by lukashergt — APPROVED** [review]

**Pull request merged**